kuka_rsi_hw_interface/src/kuka_hardware_interface_node.cpp

```diff
@@ -50,6 +50,58 @@ int main(int argc, char** argv)
 
   ros::NodeHandle nh;
 
+  std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
+  bool has_realtime = false;
+  if (realtime_file.is_open())
+  {
+    realtime_file >> has_realtime;
+  }
+  if (has_realtime)
+  {
+    const int max_thread_priority = sched_get_priority_max(SCHED_FIFO);
+    if (max_thread_priority != -1)
+    {
+      // We'll operate on the currently running thread.
+      pthread_t this_thread = pthread_self();
+
+      // struct sched_param is used to store the scheduling priority
+      struct sched_param params;
+
+      // We'll set the priority to the maximum.
+      params.sched_priority = max_thread_priority;
+
+      int ret = pthread_setschedparam(this_thread, SCHED_FIFO, &params);
+      if (ret != 0)
+      {
+        ROS_ERROR_STREAM("Unsuccessful in setting main thread realtime priority. Error code: " << ret);
+      }
+      // Now verify the change in thread priority
+      int policy = 0;
+      ret = pthread_getschedparam(this_thread, &policy, &params);
+      if (ret != 0)
+      {
+        ROS_ERROR_STREAM("Couldn't retrieve real-time scheduling paramers");
```

Comment:
```suggestion
        ROS_ERROR_STREAM("Couldn't retrieve real-time scheduling parameters");
```
copy paste of a typo present in UR as well

Reply:
Thanks for catching it; taken care of in [6eb8549](https://github.com/ros-industrial/kuka_experimental/pull/257/commits/6eb85497e0f8a4d654cc285cc8b5a4bec544e850)